packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.scss

```diff
@@ -15,6 +15,7 @@
   box-sizing: border-box;
   display: flex;
   position: relative;
+  border: 1px solid transparent;
```

Comment:
Second vote for the pseudo-element. @aleksandrjPersonal previously worked on making the focus border mixin support adding the border even when the element is not in focus. Would be good to have several mixins to handle borders: always visible, visible for high contrast, and visible when focused using the keyboard.